### PR TITLE
refactor: Replace `consola` with `pino`

### LINKED ISSUE #124: [REFACTOR] Improve logging
As a developer and operator of my auth-service, I would like to see cleaner logging to get more information about what is happening. The current logging is very vague and only outputs logs when errors occur (as it seems).

Therefore, it would be desirable to improve the logging and to offer a possibility to transport the logs to an external service. Winston, for example, would be suitable for this, giving you the option of only outputting logs, saving them in a file or transferring them to external services or our own database.

### PR DESCRIPTION
**Pull request by tklein1801**
So far, the options for customizing in logging are very limited or only log-level and activation can be set. It should be discussed which aspects of the logger should be configurable by the user

closes #124

### CODE REVIEW

**Comment by tklein1801** [comment]
#124

**Comment by tklein1801** [comment]
@Bekacru just resolved the merge conflict. Do you mind running the CI again?

**Comment by Bekacru** [comment]
Sure. Also please remove pkg-lock.json and the pnpm lock file too. And remove any new deps since we're not using them.

**Comment by tklein1801** [comment]
Sure. Done.

**Comment by tklein1801** [comment]
Sure. Should be it.

**Comment by Bekacru** [comment]
My current concern is that this will remove the nice greenish success message in the CLI for migrate and generate. Maybe we could have the CLI use the consola logger directly instead, since it can't be configured. If you could make this change, this will be good to go!

**Comment by tklein1801** [comment]
Sure, let me look at that. Currently, however, Consola should already be used because, as you wrote, an unconfigured logger is imported from better-auth and therefore the standard handler (Consola) should be used.

I will check the behavior again and make changes if necessary.

File: packages/cli/src/commands/generate.ts [hunk 1]
```diff
@@ -124,7 +124,7 @@ export async function generateAction(opts: any) {
 			options.output || path.join(cwd, schema.fileName),
 			schema.code,
 		);
-		logger.success(`🚀 Schema was generated successfully!`);
+		logger.info(`🚀 Schema was generated successfully!`);
```
Comment (Bekacru) [inline packages/cli/src/commands/generate.ts]: Here it should be `success` instead

File: packages/cli/src/commands/generate.ts [hunk 2]
```diff
@@ -97,7 +97,7 @@ export async function generateAction(opts: any) {
 			} else {
 				await fs.appendFile(path.join(cwd, schema.fileName), schema.code);
 			}
-			logger.success(
```
Comment (Bekacru) [inline packages/cli/src/commands/generate.ts]: same here

File: packages/cli/src/commands/generate.ts [hunk 3]
```diff
@@ -152,7 +152,7 @@ export async function generateAction(opts: any) {
 		options.output || path.join(cwd, schema.fileName),
 		schema.code,
 	);
-	logger.success(`🚀 Schema was generated successfully!`);
+	logger.info(`🚀 Schema was generated successfully!`);
```
Comment (Bekacru) [inline packages/cli/src/commands/generate.ts]: here

File: packages/cli/src/commands/migrate.ts [hunk 4]
```diff
@@ -102,7 +102,7 @@ export async function migrateAction(opts: any) {
 	spinner?.start("migrating...");
 	await runMigrations();
 	spinner.stop();
-	logger.success("🚀 migration was completed successfully!");
+	logger.info("🚀 migration was completed successfully!");
```
Comment (Bekacru) [inline packages/cli/src/commands/migrate.ts:105]: here as well

File: packages/cli/src/utils/get-config.ts [hunk 5]
```diff
@@ -140,7 +140,6 @@ export async function getConfig({
 							config.auth?.options || config.default?.options || null;
 						if (!configFile) {
 							logger.error("[#better-auth]: Couldn't read your auth config.");
-							logger.break();
```
Comment (Bekacru) [inline packages/cli/src/utils/get-config.ts:144]: you can change this to just empty `console.log("")` instead

**Review by Bekacru — CHANGES_REQUESTED** [review]

**Comment by tklein1801** [comment]
Sorry, I misunderstood you. How about the following solution?
We reintroduce the log level `success` for Consolas to make it possible to output successful messages, but if a custom logger is implemented, success is replaced by info as the log level, as success does not technically exist as a log level.

**Comment by tklein1801** [comment]
I made a mistake with the log-Level the get-config.ts file inside the cli Package. 

Will fix later

**Comment by tklein1801** [comment]
Done. Everything should be set now

**Comment by Bekacru** [comment]
hey @tklein1801 please update the snapshot

**Comment by tklein1801** [comment]
Hi @Bekacru, I've already done this. I am still looking at the other two failed tests

**Comment by tklein1801** [comment]
**Test `oauth.test.ts` `oauth2 - should handle server error during OAuth flow`**

Runs into `cookie-utils.ts` and stopps right here

I think if the process cannot continue due to an error, an error should be thrown.
```ts
if (!name || value === undefined) {
  console.warn(`Malformed cookie: ${cookieString}`);
  return;
}
```

**Comment by tklein1801** [comment]
Well, I'll be honest... It took me a while to work out where the error was. 
Build and test are succeeding in local...

**Pull request merged**